common/djangoapps/enrollment/views.py

```diff
@@ -345,12 +363,23 @@ def post(self, request):
             )
 
         try:
-            response = api.add_enrollment(user, unicode(course_id))
+            # Check to see if the enrollment mode is something other than honor. If so, check to see if the user
+            # is enrolled, and if so, perform an upgrade instead of creating a new enrollment.
+            enrollment = {}
+            http_success_status = status.HTTP_200_OK
+            if mode is not 'honor':
```

Comment:
These blocks are a little confusing. It might help if the condition was inverted (e.g. `if mode == 'honor'`). Also, as written, an update call will be made even if the user is already enrolled in the mode being requested. This may pose problems for analytics if multiple enrollment events are fired.